### ISSUE TITLE
Duplicate Welcome Message on Page Launch

### ISSUE THREAD

**Issue by Sreyeesh** [Sreyeesh/sreyeesh-devops-terminal#5]
### **New Issue: Duplicate Welcome Message on Page Launch**

---

#### **Title**  
Duplicate "Welcome to my Terminal Portfolio" message appears on page launch  

---

#### **Description**  
When launching the terminal portfolio, the welcome message and prompt are displayed twice, leading to redundancy and a poor user experience.  

---

#### **Steps to Reproduce**  
1. Open the terminal portfolio page in the browser.  
2. Observe the initial output.  
3. Notice that the welcome message and command prompt are displayed **twice**.  

---

#### **Expected Behavior**  
The welcome message and prompt should appear **only once** when the page is loaded.

---

#### **Actual Behavior**  
The welcome message and prompt are displayed **twice**.

---

#### **Root Cause Hypothesis**  
- The `initializeTerminal()` function may be called multiple times, possibly due to:  
   - Duplicate event listeners for `DOMContentLoaded`.  
   - Code being executed twice unintentionally.  

---

#### **Solution Steps**  
1. **Check for Multiple Event Listeners**:  
   Ensure that `DOMContentLoaded` is not being attached multiple times.  

   Example fix:  
   ```javascript
   document.addEventListener("DOMContentLoaded", () => {
     if (window.terminalInitialized) return; // Prevent duplicate initialization
     window.terminalInitialized = true;

     const terminal = document.getElementById("terminal");
     initializeTerminal();
   });
   ```

2. **Debug Initialization Call**:  
   Add a `console.log` or breakpoint inside `initializeTerminal()` to confirm if it runs more than once.

---

#### **Tasks**  
1. [ ] Confirm whether `initializeTerminal()` runs twice.  
2. [ ] Check for duplicate event listeners or redundant function calls.  
3. [ ] Apply a guard clause to prevent multiple initializations.  
4. [ ] Test to ensure the welcome message appears only once.

---

#### **Acceptance Criteria**  
- The welcome message and prompt are displayed only once on page load.  
- No duplicate initializations occur.  

---

#### **Priority**  
High  

---

Would you like help fixing this issue directly? Let me know! 🚀
